src/Raven.Server/Documents/TcpHandlers/SubscriptionDocumentsFetcher.cs

```diff
@@ -147,6 +147,77 @@ public class SubscriptionDocumentsFetcher
             }
         }
 
+        public IEnumerable<(Document Doc, Exception Exception)> GetDocumentsToSendWithTimeLimit(DocumentsOperationContext docsContext,
```

Comment:
Can we add the timeLimit to the `GetDataToSend` but when it is `null` then not apply the logic? This will avoid a lot of code redundancy I think?